downloader/src/main/kotlin/Downloader.kt

```diff
@@ -321,6 +361,25 @@ class Downloader {
         return DownloadResult(startTime, outputDirectory, sourceArtifact = target.sourceArtifact)
     }
 
+    private fun downloadPomArtifact(target: Package, outputDirectory: File): DownloadResult {
+        val pomUrl = target.binaryArtifact.url.replaceAfterLast('.', "pom")
+
+        log.info {
+            "Trying to download POM artifact for '${target.id}' from '$pomUrl'..."
+        }
+
+        val startTime = Instant.now()
+
+        try {
+            val pomName = pomUrl.substringAfterLast('/')
+            File(outputDirectory, pomName).writeBytes(URL(pomUrl).readBytes())
+        } catch (e: FileNotFoundException) {
+            throw DownloadException("Failed to download the Maven POM for '${target.id}'.")
+        }
+
+        return DownloadResult(startTime, outputDirectory, sourceArtifact = target.binaryArtifact.copy(url = pomUrl))
```

Comment:
This would re-use the hash of the binary artifact, you should calculate the hash of the downloaded file above instead. I think moving the return into the try block would make that easier.

Reply:
Again, good catch. Done.